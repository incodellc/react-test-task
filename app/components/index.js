export { PageNotFound } from './PageNotFound';
export { Header } from './Header';
export { TickerInfo } from './TickerInfo';
export { FetchIntervalSelector } from './FetchIntervalSelector';
