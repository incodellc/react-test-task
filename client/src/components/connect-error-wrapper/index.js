import ConnectErrorWrapper from './ConnectErrorWrapper';
export default ConnectErrorWrapper;