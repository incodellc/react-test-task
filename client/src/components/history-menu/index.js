import HistoryMenu from './HistoryMenu';
export default HistoryMenu;