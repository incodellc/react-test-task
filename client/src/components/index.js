import InfoBar from './info-bar';
import Layout from './layout';
import ConnectErrorWrapper from './connect-error-wrapper';
import TickerList from './ticker-list';
import TickerListItem from './ticker-list-item';

export { InfoBar, Layout, ConnectErrorWrapper, TickerList, TickerListItem };