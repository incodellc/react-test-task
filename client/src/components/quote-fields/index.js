import QuoteFields from './QuoteFields';
export default QuoteFields;