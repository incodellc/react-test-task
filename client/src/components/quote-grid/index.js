import QuotesGrid from './QuotesGrid';
export default QuotesGrid;