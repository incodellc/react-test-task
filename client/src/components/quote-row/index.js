import QuoteRow from './QuoteRow';
export default QuoteRow;