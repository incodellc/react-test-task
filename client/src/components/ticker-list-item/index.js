import TickerListItem from './TickerListItem';
export default TickerListItem;