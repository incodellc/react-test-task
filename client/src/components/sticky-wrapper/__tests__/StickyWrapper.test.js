import React from 'react';
import { StickyWrapper } from '../../../components';
import { cleanup, render } from '@testing-library/react';

describe('Check StickyWrapper component', () => {
    afterEach(cleanup);
    
    const SW = (
        <StickyWrapper
            stickyIndex={1}
        >
            <div data-testid='sticky-wrapper-target'>Content 1</div>
            <div data-testid='sticky-wrapper-target'>Content 2</div>
            <div data-testid='sticky-wrapper-target'>Content 3</div>
        </StickyWrapper>
    );

    it('component renders correctly', () => {
        const { asFragment } = render(SW);
        expect(asFragment()).toMatchSnapshot();
    });

    it('component should contain all wrapped elements', () => {
        const { getByTestId, getAllByTestId } = render(SW);
        expect(getByTestId('sticky-wrapper')).toContainElement(getAllByTestId('sticky-wrapper-target')[0]);
        expect(getAllByTestId('sticky-wrapper-target').length).toEqual(3);
    });

    it('sticky element should be in the right place due to value of stickyIndex', () => {
        const { getByTestId, getByText } = render(SW);
        expect(getByTestId('sticky-wrapper-fragment')).toContainElement(getByText('Content 2'));
    });
});