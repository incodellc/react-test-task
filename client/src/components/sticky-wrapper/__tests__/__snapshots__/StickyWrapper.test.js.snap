// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check StickyWrapper component component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="ui basic segment stickyWrapper"
    data-testid="sticky-wrapper"
  >
    <div>
      <div
        class="ui basic segment"
      >
        <div
          data-testid="sticky-wrapper-target"
        >
          Content 1
        </div>
      </div>
      <div
        class=""
        data-testid="sticky-wrapper-fragment"
      >
        <div />
        <div
          class="ui sticky"
        >
          <div
            class="ui basic segment"
          >
            <div
              data-testid="sticky-wrapper-target"
            >
              Content 2
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui basic segment"
      >
        <div
          data-testid="sticky-wrapper-target"
        >
          Content 3
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
